nautobot_netbox_importer/diffsync/nautobot.py

```diff
@@ -0,0 +1,432 @@
+"""Nautobot DiffSync Importer."""
+
+from typing import Any
+from typing import Dict
+from typing import Iterable
+from typing import Mapping
+from typing import MutableMapping
+from typing import NamedTuple
+from typing import Optional
+from typing import Set
+from typing import Tuple
+from typing import Type
+from uuid import UUID
+
+from diffsync import DiffSyncModel
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from django.db.models import Max
+from nautobot.core.utils.lookup import get_model_from_name
+
+from .base import EMPTY_VALUES
+from .base import INTERNAL_TYPE_TO_ANNOTATION
+from .base import REFERENCE_INTERNAL_TYPES
+from .base import BaseAdapter
+from .base import ContentTypeStr
+from .base import DjangoFieldDoesNotExist
+from .base import DjangoModelMeta
+from .base import FieldName
+from .base import GenericForeignKey
+from .base import GenericForeignValue
+from .base import InternalFieldTypeStr
+from .base import NautobotBaseModel
+from .base import NautobotBaseModelType
+from .base import NautobotField
+from .base import PydanticField
+from .base import RecordData
+from .base import Uid
+from .base import logger
+
+# Helper to determine the import order of models.
+# Due to dependencies among Nautobot models, certain models must be imported first to ensure successful `instance.save()` calls without errors.
+# Models listed here take precedence over others, which are sorted by the order they're introduced to the importer.
+# Models listed, but not imported, here will be ignored.
+# Obsoleted models can be kept here to ensure backward compatibility.
+IMPORT_ORDER: Iterable[ContentTypeStr] = (
+    "extras.customfield",
+    "extras.customfieldchoice",
+    "extras.status",
+    "extras.role",
+    "dcim.locationtype",
+    "dcim.location",
+    "tenancy.tenantgroup",
+    "tenancy.tenant",
+    "users.user",
+    "circuits.circuit",
+    "circuits.circuittermination",
+    "circuits.circuittype",
+    "circuits.provider",
+    "circuits.providernetwork",
+    "dcim.devicetype",
+    "dcim.consoleport",
+    "dcim.consoleporttemplate",
+    "dcim.device",
+    "dcim.devicebay",
+    "dcim.frontport",
+    "dcim.frontporttemplate",
+    "dcim.interface",
+    "dcim.interfacetemplate",
+    "dcim.powerfeed",
+    "dcim.poweroutlet",
+    "dcim.poweroutlettemplate",
+    "dcim.powerpanel",
+    "dcim.powerport",
+    "dcim.powerporttemplate",
+    "dcim.rack",
+    "dcim.rearport",
+    "dcim.rearporttemplate",
+    "dcim.cable",
+    "ipam.prefix",
```

Comment:
Do you need to add the remainder of IPAM?

Reply:
Not necessary to add other models for now, this list is just to precede those models, that needs to be imported between others in a specific order.